src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -83,10 +82,48 @@ public void storeMessage(Message message) {
 
     return messages;
   }
-  /** Returns the total number of messages for all users. */
-  public int getTotalMessageCount(){
+
+  /**
+   * Returns the total number of messages for all users.
+   */
+  public int getTotalMessageCount() {
     Query query = new Query("Message");
     PreparedQuery results = datastore.prepare(query);
     return results.countEntities(FetchOptions.Builder.withLimit(1000));
   }
+
+  /**
+   * Stores the user input for the given marker location on the map.
+   */
+  public List<UserMarker> getMarkers() {
+    List<UserMarker> markers = new ArrayList<>();
+
+    Query query = new Query("UserMarker");
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      try {
+        double lat = (double) entity.getProperty("lat");
+        double lng = (double) entity.getProperty("lng");
+        String content = (String) entity.getProperty("content");
+
+        UserMarker marker = new UserMarker(lat, lng, content);
+        markers.add(marker);
+      } catch (Exception e) {
+        System.err.println("Error reading marker.");
+        System.err.println(entity.toString());
+        e.printStackTrace();
+      }
+    }
+    return markers;
+  }
+
+  public void storeMarker(UserMarker marker) {
```

Comment:
Can you please explain what does this part of the code does to me?

Reply:
This code stores the markers created on the map and loads it when it is open